ocrd_models/ocrd_models/ocrd_page.py

```diff
@@ -73,3 +87,101 @@ def to_xml(el):
                 NAMESPACES['page']
             ))
     return '<?xml version="1.0" encoding="UTF-8"?>\n' + sio.getvalue()
+
+class OcrdPageExt():
+    """
+    Helper class for translating between generateDS and etree API and
+    implementing Region iterator.
+
+    Nomenclature:
+    - "obj" refers to a type class instance generated by generateDS
+    - "el" is an etree element
+    """
+
+    @classmethod
+    def monkey_patch(cls, pcgts):
+        pcgts.ocrd_ext = cls(pcgts)
+        return pcgts
+
+    def __init__(self, root_obj):
+        self.root_obj = root_obj
+        self.mapping_obj2el = None
+        self.mapping_el2obj = None
+        self.root_el = None
+        self.update_mappings()
+
+    def update_mappings(self):
+        """
+        Needs to be called whenever the PAGE document changes and this class
+        shall be used
+        """
+        t0 = perf_counter()
+        self.mapping_obj2el = {}
+        self.root_el = self.root_obj.to_etree(None, name_='PcGts', mapping_=self.mapping_obj2el)
+        self.mapping_el2obj = dict(((v, k) for k, v in self.mapping_obj2el.items()))
+        LOG.debug("OcrdPageExt.update_mappings took %ss", perf_counter()-t0)
```

Comment:
Curious whether you tested how this affects performance - thinking of e.g. large documents like newspapers or pages with polygonal glyph segmentation that can have thousands of elements.

Reply:
`update_mappings` will degrade performance, I cannot say by how much. But certainly enough not to call it every time the document changes.